### PR TITLE
ci: use prom package from sig-instrumentation

### PR DESCRIPTION
**Pull request by sthaha**

### CODE REVIEW

File: test/e2e/framework/assertions.go [hunk 1]
```diff
@@ -124,3 +132,125 @@ func (f *Framework) GetResourceWithRetry(t *testing.T, name, namespace string, o
...
+
+	vec, err := res.Vector()
+	if res.Err != nil {
+		t.Errorf("error converting to scalar: %v", err)
+	}
+
+	for _, v := range vec {
+		s, ok := expected[v.Metric.String()]
+		if !ok {
+			t.Errorf("Didn't expect this label combo: %v\n", v.Metric.String())
```
Comment (fpetkovski) [inline test/e2e/framework/assertions.go:182]:
```suggestion
			t.Errorf("Didn't expect this labelset: %v\n", v.Metric.String())
```

File: test/e2e/framework/assertions.go [hunk 2]
```diff
@@ -3,14 +3,22 @@ package framework
...
+	"io"
+	"net/http"
 	"testing"
 	"time"
 
-	appsv1 "k8s.io/api/apps/v1"
+	// _ "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/promql"
 
+	// "github.com/rhobs/monitoring-stack-operator/test/e2e/framework/prom"
```
Comment (fpetkovski) [inline test/e2e/framework/assertions.go:14]: Do we need this line?

**Review by fpetkovski — COMMENTED** [review]
Left a few small nits. Also the operator binary is added to the commit.

File: test/e2e/main_test.go [hunk 3]
```diff
@@ -50,7 +50,21 @@ func main(m *testing.M) int {
...
 	}
 
-	return m.Run()
+	exitCode := m.Run()
+
+	tests := []testing.InternalTest{{
+		Name: "NoReconcilationErrors",
+		F: func(t *testing.T) {
+			f.AssertNoReconcileErrors(t)
+		},
```
Comment (fpetkovski) [inline test/e2e/main_test.go:59]: Wouldn't this also work?

```suggestion
		F: f.AssertNoReconcileErrors,
```

Comment (sthaha) [reply test/e2e/main_test.go:59]: aha! good catch :D

**Review by fpetkovski — APPROVED** [review]

**Pull request merged**